src/us/kbase/test/auth2/MongoController.java

```diff
@@ -0,0 +1,143 @@
+package us.kbase.test.auth2;
+
+import static us.kbase.common.test.controllers.ControllerCommon.checkExe;
+import static us.kbase.common.test.controllers.ControllerCommon.findFreePort;
+import static us.kbase.common.test.controllers.ControllerCommon.makeTempDirs;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+
+import com.github.zafarkhaja.semver.Version;
+import com.mongodb.client.MongoClients;
+import org.apache.commons.io.FileUtils;
+import org.bson.Document;
+
+
+/** Q&D Utility to run a Mongo server for the purposes of testing from
+ * Java.
+ * @author gaprice@lbl.gov, sijiex@lbl.gov
+ *
+ */
+public class MongoController {
+
+    private final static String DATA_DIR = "data";
+
+    private final static List<String> tempDirectories =
+            new LinkedList<String>();
+    static {
+        tempDirectories.add(DATA_DIR);
+    }
+
+    private final static Version MONGO_DB_3 =
+            Version.forIntegers(3,6,23);
+
+    private final static Version MONGO_DB_7 =
+            Version.forIntegers(7,0,4);
+
+    private final Path tempDir;
+    private final int port;
+    private Process mongo;
+
+    public MongoController(
+            final String mongoExe,
+            final Path rootTempDir)
+            throws Exception {
+        this(mongoExe, rootTempDir, false);
+    }
+
+    public MongoController(
+            final String mongoExe,
+            final Path rootTempDir,
+            final boolean useWiredTiger)
+            throws Exception {
+        checkExe(mongoExe, "mongod server");
+        tempDir = makeTempDirs(rootTempDir, "MongoController-", tempDirectories);
+        port = findFreePort();
+
+        List<String> command = getCommand(mongoExe, useWiredTiger);
+        mongo = startProcess(command);
+        Version dbVer = getMongoDBVer();
+        if (dbVer.greaterThan(MONGO_DB_3)) {
+            destroy(false);
+            command = getCommand(mongoExe, useWiredTiger, MONGO_DB_7);
+            mongo = startProcess(command);
+        }
+    }
+
```

Comment:
This means that if the server is mongo 3 it'll never get started. Also, you should find out when no-journal was actually removed and hardcode that version

Reply:
it will not, because it will just use the first process initiated.

Reply:
Oh, I was thinking for the first time you ran the process you'd just do `mongod --version` and parse the result

Reply:
Which I think would be a better approach than starting the server

Reply:
Updated!

Reply:
:+1: